bootloader/src/main.c

```diff
@@ -414,8 +417,19 @@ int main(void)
             // Initialize user application's Stack Pointer
             __set_MSP(*(__IO uint32_t*) ApplicationAddress);
 
-            // Set IWDG Timeout to 5 secs based on platform specific system flags
-            IWDG_Reset_Enable(5 * TIMING_IWDG_RELOAD);
+            uint8_t disable_iwdg = 0;
+#ifdef CHECK_FIRMWARE
+            // Pre-0.7.0 firmwares were expecting IWDG flag to be set in the DCT, now it's stored in
+            // the backup registers. As a workaround, we disable IWDG if an older firmware is detected
+            const uint16_t module_ver = get_main_module_version();
+            if (module_ver >= 0 && module_ver < SYSTEM_MODULE_VERSION_0_7_0_RC1) {
```

Comment:
Yep. Ideally, bootloader should always check integrity of a newly flashed firmware with all its dependencies, and then save a flag to the DCT to avoid checking it until next update. This will allow bootloader to enter DFU mode instantly in case of a corrupted firmware, without relying on IWDG (which, in turn, ideally should be kept enabled all the time).